Packs/Illumio/Integrations/IllumioCore/IllumioCore_test.py

```diff
@@ -38,3 +55,224 @@ def test_test_module(requests_mock, mock_client):
     assert test_module(mock_client) == "ok"
+
+
+@pytest.mark.parametrize("args, err_msg", [
+    ({"port": "", "protocol": "tcp", 'start_time': '2022-07-17T12:58:33.528Z', 'end_time': "2022-07-18T12:58:33.529Z",
+      'policy_decisions': ['potentially_blocked']},
+     "{} is a required parameter. Please provide correct value.".format('port')),
+    ({"port": "dummy", "protocol": "tcp", 'start_time': '2022-07-17T12:58:33.528Z',
+      'end_time': "2022-07-18T12:58:33.529Z", 'policy_decisions': ['potentially_blocked']},
+     '"dummy" is not a valid number'),
+    ({"port": -300, "protocol": "tcp", 'start_time': '2022-07-17T12:58:33.528Z',
+      'end_time': "2022-07-18T12:58:33.529Z", 'policy_decisions': ['potentially_blocked']},
+     "{} is an invalid value for port. Value must be in 1 to 65535.".format(-300)),
+    ({"port": 8443, "protocol": "tcp", 'start_time': '2022-07-17T12:58:33.528Z',
+      'end_time': "2022-07-18T12:58:33.529Z", 'policy_decisions': ['dummy']},
+     "Invalid value for {}. Possible comma separated values are {}.".format('policy_decisions',
+                                                                            VALID_POLICY_DECISIONS)),
+    ({"port": 8443, "protocol": "dummy", 'start_time': '2022-07-17T12:58:33.528Z',
+      'end_time': "2022-07-18T12:58:33.529Z", 'policy_decisions': ['potentially_blocked']},
+     "{} is an invalid value for {}. Possible values are: {}.".format('dummy', 'protocol', VALID_PROTOCOLS))
+])
+def test_illumio_traffic_analysis_command_for_invalid_arguments(args, err_msg, mock_client):
+    """
+    Test case scenario for execution of illumio-traffic-analysis-command when invalid argument provided.
+
+    Given:
+        - command arguments for illumio_traffic_analysis_command
+    When:
+        - Calling `illumio_traffic_analysis_command` function
+    Then:
+        - Returns a valid error message
+    """
+    from IllumioCore import illumio_traffic_analysis_command
+    with pytest.raises((ValueError, IllumioException, InvalidPortException, InvalidSingleSelectException,
+                        InvalidMultiSelectException)) as err:
+        illumio_traffic_analysis_command(mock_client, args)
+    assert str(err.value) == err_msg
+
+
+def test_illumio_traffic_analysis_success(requests_mock, mock_client):
+    """
+    Test case scenario for successful execution of illumio_traffic_analysis_command function.
+
+    Given:
+        - command arguments for illumio_traffic_analysis
+    When:
+        - Calling `illumio_traffic_analysis_Command` function
+    Then:
+        - Returns a valid output
+    """
+    from IllumioCore import illumio_traffic_analysis_command
+
+    args = {'port': 8443, 'protocol': "tcp", 'policy_decisions': 'potentially_blocked',
+            'start_time': "2022-07-17T12:58:33.528Z", 'end_time': "2022-07-18T12:58:33.529Z"}
+
+    json_data = util_load_json(os.path.join(os.path.dirname(os.path.realpath(__file__)),
```

Comment:
Don't repeat this path join every time. Create a constant with the absolute path to the data directory.

Reply:
Okay